openassessment/xblock/staff_info_mixin.py

```diff
@@ -208,58 +208,59 @@ def render_student_info(self, data, suffix=''):  # pylint: disable=W0613
         """
         Renders all relative information for a specific student's workflow.
 
-        Given a student's ID, we can render a staff-only section of the page
+        Given a student's username, we can render a staff-only section of the page
         with submissions and assessments specific to the student.
 
         Must be course staff to render this view.
 
         """
         try:
-            student_id = data.params.get('student_id', '')
-            path, context = self.get_student_info_path_and_context(student_id)
+            student_username = data.params.get('student_username', '')
+            path, context = self.get_student_info_path_and_context(student_username)
             return self.render_assessment(path, context)
 
         except PeerAssessmentInternalError:
             return self.render_error(self._(u"Error finding assessment workflow cancellation."))
 
-    def get_student_info_path_and_context(self, student_id):
+    def get_student_info_path_and_context(self, student_username):
         """
         Get the proper path and context for rendering the the student info
         section of the staff debug panel.
 
         Args:
-            student_id (unicode): The ID of the student to report.
+            student_username (unicode): The username of the student to report.
 
         """
         submission_uuid = None
         submission = None
         assessment_steps = self.assessment_steps
 
-        if student_id:
-            student_item = self.get_student_item_dict()
-            student_item['student_id'] = student_id
-
-            # If there is a submission available for the requested student, present
-            # it. If not, there will be no other information to collect.
-            submissions = submission_api.get_submissions(student_item, 1)
-
-            if submissions:
-                submission_uuid = submissions[0]['uuid']
-                submission = submissions[0]
-
-                if 'file_key' in submission.get('answer', {}):
-                    file_key = submission['answer']['file_key']
-
-                    try:
-                        submission['image_url'] = file_api.get_download_url(file_key)
-                    except file_api.FileUploadError:
-                        # Log the error, but do not prevent the rest of the student info
-                        # from being displayed.
-                        msg = (
-                            u"Could not retrieve image URL for staff debug page.  "
-                            u"The student ID is '{student_id}', and the file key is {file_key}"
-                        ).format(student_id=student_id, file_key=file_key)
-                        logger.exception(msg)
+        if student_username:
```

Comment:
Nitpick, but I think you could avoid some of this indentation. For example, line 242 does not have to be nested, since anonymous_user_id will be None if 238 is not true, and 247 could be brought out of the nested if statement, since submissions will be None if anonymous_user_id is None.

Not a big deal, but may read a little more cleanly.